### PR TITLE
Code cleanup (golangci-lint)

### PR DESCRIPTION
**Pull request by gavinelder**
# Summary

The following is an attempt to reduce the amount of linting errors the project has using https://github.com/golangci/golangci-lint

### Before

```
main.go:48:2: `httpIP` is unused (deadcode)
	httpIP   string
	^
main.go:49:2: `httpPort` is unused (deadcode)
	httpPort string
	^
main.go:67:2: `httpReadLimit` is unused (deadcode)
	httpReadLimit int64 = 2 * (1024 * 1024)
	^
main.go:400:6: `BestDomain` is unused (deadcode)
func BestDomain() string {
     ^
handlers.go:134:19: Error return value of `config.UpdateInfo` is not checked (errcheck)
	config.UpdateInfo(func(i *Info) error {
	                 ^
handlers.go:181:21: Error return value of `config.UpdateInfo` is not checked (errcheck)
			config.UpdateInfo(func(i *Info) error {
			                 ^
handlers.go:209:19: Error return value of `config.UpdateInfo` is not checked (errcheck)
	config.UpdateInfo(func(i *Info) error {
	                 ^
handlers.go:283:19: Error return value of `config.UpdateUser` is not checked (errcheck)
	config.UpdateUser(user.ID, func(u *User) error {
	                 ^
handlers.go:468:16: Error return value of `f.WriteString` is not checked (errcheck)
		f.WriteString(errstr)
		             ^
main.go:103:11: Error return value of `cli.Parse` is not checked (errcheck)
	cli.Parse(os.Args[1:])
	         ^
main.go:327:17: Error return value of `tc.SetKeepAlive` is not checked (errcheck)
	tc.SetKeepAlive(true)
	               ^
main.go:328:23: Error return value of `tc.SetKeepAlivePeriod` is not checked (errcheck)
	tc.SetKeepAlivePeriod(10 * time.Minute)
	                     ^
web.go:191:22: SA4017: WithContext is a pure function but its return value is ignored (staticcheck)
				r = r.WithContext(samlsp.WithToken(r.Context(), token))
				                 ^
handlers.go:39:2: S1023: redundant `return` statement (gosimple)
	return
	^
handlers.go:146:2: S1023: redundant `return` statement (gosimple)
	return
	^
handlers.go:220:2: S1023: redundant `return` statement (gosimple)
	return
	^
config.go:280:2: S1011: should replace loop with `profiles = append(profiles, c.Profiles...)` (gosimple)
	for _, p := range c.Profiles {
	^
config.go:415:2: S1011: should replace loop with `users = append(users, c.Users...)` (gosimple)
	for _, u := range c.Users {
	^
utils.go:65:61: S1030: should use script.String() instead of string(script.Bytes()) (gosimple)
	output, err := exec.CommandContext(ctx, "/bin/bash", "-c", string(script.Bytes())).CombinedOutput()
	                                                           ^

```

### Current
```
handlers.go:133:19: Error return value of `config.UpdateInfo` is not checked (errcheck)
	config.UpdateInfo(func(i *Info) error {
	                 ^
handlers.go:179:21: Error return value of `config.UpdateInfo` is not checked (errcheck)
			config.UpdateInfo(func(i *Info) error {
			                 ^
handlers.go:207:19: Error return value of `config.UpdateInfo` is not checked (errcheck)
	config.UpdateInfo(func(i *Info) error {
	                 ^
handlers.go:280:19: Error return value of `config.UpdateUser` is not checked (errcheck)
	config.UpdateUser(user.ID, func(u *User) error {
	                 ^
handlers.go:465:16: Error return value of `f.WriteString` is not checked (errcheck)
		f.WriteString(errstr)
		             ^
main.go:96:11: Error return value of `cli.Parse` is not checked (errcheck)
	cli.Parse(os.Args[1:])
	         ^
main.go:320:17: Error return value of `tc.SetKeepAlive` is not checked (errcheck)
	tc.SetKeepAlive(true)
	               ^
main.go:321:23: Error return value of `tc.SetKeepAlivePeriod` is not checked (errcheck)
	tc.SetKeepAlivePeriod(10 * time.Minute)
	                     ^
web.go:191:22: SA4017: WithContext is a pure function but its return value is ignored (staticcheck)
				r = r.WithContext(samlsp.WithToken(r.Context(), token))
				                 ^
utils.go:65:61: S1030: should use script.String() instead of string(script.Bytes()) (gosimple)
	output, err := exec.CommandContext(ctx, "/bin/bash", "-c", string(script.Bytes())).CombinedOutput()
	                                                           ^

```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -44,10 +44,6 @@ var (
 	// Insecure http cookies (only recommended for internal LANs/VPNs)
 	httpInsecure bool
 
-	// set based on httpAddr
-	httpIP   string
-	httpPort string
```
Comment (gavinelder) [inline main.go:49]: There two were declared on line 136 of main.go using the go short declaration syntax of `:=` so these vars are being shadowed.

File: main.go [hunk 2]
```diff
@@ -63,9 +59,6 @@ var (
 	// Let's Encrypt
 	letsencrypt bool
 
-	// HTTP read limit
-	httpReadLimit int64 = 2 * (1024 * 1024)
```
Comment (gavinelder) [inline main.go:67]: No use or reference in this to the code.

File: main.go [hunk 3]
```diff
@@ -397,10 +390,11 @@ func configureSAML() error {
...
 }
 
-func BestDomain() string {
-	domain := config.FindInfo().Domain
-	if domain != "" {
-		return domain
-	}
-	return httpHost
-}
+// The following is unused not removing just commenting out
```
Comment (jack1902) [inline main.go:393]: curious as to what the original intent of this was

Comment (gavinelder) [reply main.go:393]: Okay I found out what the original intent was by looking at https://github.com/frequencyanalytics/frequency/blob/master/main.go another project by cloud.app 

TLDR this should have been used to find the domain however it as replaced by the httpHost lookup as part of doing cookie auth. 

So this whole thing can be killed.

**Comment by gavinelder** [comment]
@jack1902 @agonbar can you please take a look at this would like to land it sooner rather than later.
